internal/transport/http_util.go

```diff
@@ -293,6 +285,26 @@ func (d *decodeState) constructHTTPErrMsg() string {
 	return strings.Join(errMsgs, "; ")
 }
 
+// constructErrMsg constructs error message to be returned in HTTP fallback mode.
+// Format: HTTP status code and its corresponding message + content-type error message.
+func constructHTTPErrMsg(httpStatus *int, contentTypeErr string) string {
```

Comment:
This looks identical to the above; if so, please make `decodeState.constructHTTPErrMsg` call this one instead of copy/pasting.

Reply:
Done!